src/main/java/org/killbill/billing/plugin/vertex/dao/VertexDao.java

```diff
@@ -54,13 +54,15 @@ public class VertexDao extends PluginDao {
     private static final Logger logger = LoggerFactory.getLogger(VertexDao.class);
     private static final String SUCCESS = "SUCCESS";
     private static final String ERROR = "ERROR";
+    private final VertexResponseDataExtractor vertexResponseDataExtractor;
 
     static {
         objectMapper.registerModule(new JavaTimeModule());
     }
 
-    public VertexDao(final DataSource dataSource) throws SQLException {
+    public VertexDao(final DataSource dataSource, final VertexResponseDataExtractor vertexResponseDataExtractor) throws SQLException {
```

Comment:
`VertexResponseDataExtractor` is just a util class that processes some data. Do we need it in the constructor?
I was thinking if it makes sense to have something like below 
```
public VertexDao(final DataSource dataSource) {
    // as before
}

public void addResponse(.....
    final ApiSuccessResponseTransactionResponseType taxResult
} {
    VertexResponseDataExtractor vertexResponseDataExtractor = new VertexResponseDataExtractor(taxResult)

  // for all details use vertexResponseDataExtractor
}

```

Reply:
Thank you for the proposal, I agree - makes sense.

Reply:
Fixed